actors/builtin/reward/reward_actor.go

```diff
@@ -142,15 +139,20 @@ func (a Actor) UpdateNetworkKPI(rt vmr.Runtime, currRealizedPower *abi.StoragePo
 
 	var st State
 	rt.State().Transaction(&st, func() interface{} {
-		newBaselinePower := a.newBaselinePower(&st)
-		st.BaselinePower = newBaselinePower
-		st.CumsumBaseline = big.Add(st.CumsumBaseline, st.BaselinePower)
-
 		st.RealizedPower = *currRealizedPower
 		st.CumsumRealized = big.Add(st.CumsumRealized, *currRealizedPower)
 
-		st.EffectiveNetworkTime = a.getEffectiveNetworkTime(&st, st.CumsumBaseline, st.CumsumRealized)
+		for i := st.LastKPIUpdate; i < rt.CurrEpoch(); i++ {
+			newBaselinePower := a.newBaselinePower(&st)
+			st.BaselinePower = newBaselinePower
+			st.CumsumBaseline = big.Add(st.CumsumBaseline, st.BaselinePower)
+
+			st.EffectiveNetworkTime = a.getEffectiveNetworkTime(&st, st.CumsumBaseline, st.CumsumRealized)
+
+			a.computePerEpochReward(&st, i, st.EffectiveNetworkTime, 1)
```

Comment:
I would think we can do this just once at the end, though I could be missing a side-effect that's important.

Reply:
Yes, doing it once at the end sgtm unless im missing something too.

Reply:
I was wrong, it has important side effects on the supply state

Reply:
Actually I'm not so sure. Since no reward is actually minted on those null rounds, updating the supply as if there was also seems wrong.